autogen/code_utils.py

```diff
@@ -228,21 +228,33 @@ def get_powershell_command():
             if result.returncode == 0:
                 return "pwsh"
 
-        except (FileNotFoundError, NotADirectoryError):
-            if WIN32:
-                logging.warning("Neither powershell nor pwsh is installed but it is a Windows OS")
-            return None
+        except (FileNotFoundError, NotADirectoryError) as e:
+            if WIN32 and isinstance(e, FileNotFoundError):
+                raise FileNotFoundError("Neither powershell.exe nor pwsh.exe is present but it is a Windows OS")
 
+            elif WIN32 and isinstance(e, NotADirectoryError):
+                raise NotADirectoryError(
+                    "PowerShell is either not installed or its path is not given properly in the environment variable PATH. Please check the path and try again."
+                )
 
-powershell_command = get_powershell_command()
+            return "sh"
```

Comment:
Same thing here, we should not be returning `sh` here. That's unspecified behavior. Should be raising error.

Reply:
Alright, I've changed it now is this okay?